src/v2/auth.rs

```diff
@@ -185,3 +176,54 @@ impl Client {
         Box::new(fres)
     }
 }
+
+// This parses a Www-Authenticate header of value Bearer, which could look something like this:
+// `Www-Authenticate: Bearer realm="https://sat-r220-02.lab.eng.rdu2.redhat.com/v2/token",service="sat-r220-02.lab.eng.rdu2.redhat.com",scope="repository:registry:pull,push"`
```

Comment:
I'd put the spec URL here for reference, move this example to the test, and just mention "which looks like the test sample".
Also, can you please turn this into a full docstring comment with `///` and title+body split?

Reply:
I acknowledge that it's pretty verbose. Since the test already has this string I can just remove it here. Alternatively we could do a docstring test. How about a doctest for this one? It seems small enough

Reply:
Interesting, I guess we can do that.
But I thought you wanted to show the whole `Www-Authenticate: Bearer realm=<foo>` thing and not just `foo`, right?

Reply:
Not necessarily. The difficulty is caused by the stuff after `Bearer ` so I'm fine leaving the the trimming out of the example.